src/lib/ast/ast.ml

```diff
@@ -239,11 +267,40 @@ let of_sharkdown ~template_markdown =
       sections
   in
 
+  let input_map = Frontmatter.input_map metadata in
+  let synthesized_sections =
+    match List.length input_map with
+    | 0 -> []
+    | _ -> (
+        match concrete_paths with
+        | Some concrete_paths ->
+            [ synthesize_import_block input_map concrete_paths ]
+        | None -> [ synthesize_unmapped_import_block input_map ])
+  in
+
+  let expanded_markdown =
```

Comment:
It might be more maintainable in the long term to contruct the markdown via `Cmarkit` constructors, but happy to keep it simple with `printf` strings.

Reply:
Totes agree on that - I did try this first, but couldn't figure out how to do it from the Cmarkit docs, which is why I then went with the naive approach here.

Reply:
Yeh, I think there are a few pain points:

 1. The name clash between a module and the constructor i.e. `Cmarkit.Block.Code_block`
 2. The need to supply a `Meta.t` to everything 
 3. The lack of functions for constructing values forcing you to use the ADT constructors directly...
 
 Happy to refactor later @mdales ^^

Reply:
I think it's also worth noting there's a tension here between doing the expansion here, vs doing it the template expansion that @patricoferris originally built. I opted for this approach as I'm keen that I can just execute a shark file during experimental development in one step rather than two, that that then necessitates this dynamic rewriting of the shark file.